### PR TITLE
Map documentation added

### PR DESCRIPTION
**Pull request by Ranray**

### CODE REVIEW

File: Soccar [Client]/src/nl/soccar/ui/GameCanvas.java [hunk 1]
```diff
@@ -20,14 +24,29 @@ public GameCanvas() {
...
+     * Method that adds a drawable to the list of drawables.
+     * 
+     * @param drawable Drawable item that needs to be added to the list of drawables.
+     */
     public final void addDrawable(Drawable drawable) {
         drawables.add(drawable);
     }
 
+    /**
+     * Method that removes drawable of the list of drawables.
```
Comment (Laugslander) [inline Soccar [Client]/src/nl/soccar/ui/GameCanvas.java]: Method that removes drawable **out** of the list of drawables.

File: Soccar [Client]/src/nl/soccar/ui/Drawable.java [hunk 2]
```diff
@@ -8,10 +8,23 @@
  */
 public interface Drawable<T> {
     
+    /**
+     * Method that updates the position of this Drawable object.
+     */
     void update();
     
+    /**
+     * Method that draws this Drawable object on the context.
```
Comment (Laugslander) [inline Soccar [Client]/src/nl/soccar/ui/Drawable.java]: Method that draws this Drawable object on the **graphics** context.

File: Soccar [Client]/src/nl/soccar/ui/Drawable.java [hunk 3]
```diff
@@ -8,10 +8,23 @@
...
     
+    /**
+     * Method that updates the position of this Drawable object.
+     */
     void update();
     
+    /**
+     * Method that draws this Drawable object on the context.
+     * 
+     * @param context The drawable context.
```
Comment (Laugslander) [inline Soccar [Client]/src/nl/soccar/ui/Drawable.java]: @param context The drawable **graphics** context.

File: Soccar [Client]/src/nl/soccar/ui/DisplayConstants.java [hunk 4]
```diff
@@ -23,7 +23,9 @@
     public static final int BOX_HEIGHT = 325;
     public static final int GOAL_WIDTH = 35;
     public static final double GOAL_HEIGHT = BOX_HEIGHT / 1.5;
-    
+    /**
+     * Map images properties
+     */
     public static final String LOCATION_TEXTURE_MOON = "resources/images/moon_texture.jpg";
     public static final String LOCATION_TEXTURE_DESERT = "resources/images/desert_texture.jpg";
     public static final String LOCATION_TEXTURE_GRASS = "resources/images/grass_texture.jpg";
```
Comment (Laugslander) [inline Soccar [Client]/src/nl/soccar/ui/DisplayConstants.java:32]: Lege regel toevoegen na:
public static final String LOCATION_TEXTURE_GRASS = "resources/images/grass_texture.jpg";

**Review by Laugslander — COMMENTED** [review]
Toegevoegd commentaar in orde. Ik heb nog een aantal kleine (stijl)fouten gemarkeerd.

File: Soccar [Client]/src/nl/soccar/ui/GameCanvas.java [hunk 5]
```diff
@@ -20,14 +24,29 @@ public GameCanvas() {
...
+     * 
+     * @param drawable Drawable item that needs to be removed from the list.
```
Comment (bras1223) [inline Soccar [Client]/src/nl/soccar/ui/GameCanvas.java]: @param drawable Drawable item that needs to be removed from the **drawable** list.

File: Soccar [Client]/src/nl/soccar/ui/GameCanvas.java [hunk 6]
```diff
@@ -20,14 +24,29 @@ public GameCanvas() {
...
+     */
     public final void removeDrawable(Drawable drawable) {
         drawables.remove(drawable);
     }
 
+    /**
+     * Method that gets the list of drawable items of this GameCanvas.
+     * 
+     * @return List of drawables.
```
Comment (bras1223) [inline Soccar [Client]/src/nl/soccar/ui/GameCanvas.java]: @return **unmodifiable** List of drawables.

File: Soccar [Client]/src/nl/soccar/ui/input/Keyboard.java [hunk 7]
```diff
@@ -22,16 +27,32 @@ private Keyboard() {
...
+     * 
+     * @param code The keycode that needs to be added to the pressedKeys list.
+     */
     public void setKeyPressed(KeyCode code) {
         if (!pressedKeys.contains(code)) {
             pressedKeys.add(code);
         }
     }
 
+    /**
```
Comment (bras1223) [inline Soccar [Client]/src/nl/soccar/ui/input/Keyboard.java:51]: Removed of or from?

File: Soccar [Client]/src/nl/soccar/ui/input/Keyboard.java [hunk 8]
```diff
@@ -22,16 +27,32 @@ private Keyboard() {
         pressedKeys = new ArrayList<>();
     }
 
+    /**
+     * Method that checks if the given key is pressed.
+     * 
+     * @param code The keycode that needs to be checked.
+     * @return
```
Comment (bras1223) [inline Soccar [Client]/src/nl/soccar/ui/input/Keyboard.java]: - Method that checks if the given key **is already present in pressedKeys list**.

@return **boolean True if already pressed, false if not in pressedKeys list.**

File: Soccar [Client]/src/nl/soccar/ui/input/Keyboard.java [hunk 9]
```diff
@@ -12,6 +12,11 @@
 
     private static final Keyboard instance = new Keyboard();
 
+    /**
+     * Method that gets this Keyboard object.
```
Comment (bras1223) [inline Soccar [Client]/src/nl/soccar/ui/input/Keyboard.java]: - Method that gets this **static instance of the** Keyboard object.

File: Soccar [Client]/src/nl/soccar/util/MapUtilities.java [hunk 10]
```diff
@@ -9,13 +9,23 @@
  */
 public final class MapUtilities {
 
+    /**
+     * Method that calculates and returns the rectangle of the left goal of the map.
+     * 
+     * @return The rectangle of the the left goal of the map.
```
Comment (bras1223) [inline Soccar [Client]/src/nl/soccar/util/MapUtilities.java]: @return **Rectangle** The rectangle of the the left goal of the map.

File: Soccar [Client]/src/nl/soccar/util/MapUtilities.java [hunk 11]
```diff
@@ -9,13 +9,23 @@
...
+     */
     public static Rectangle getLeftGoal() {
         double goalPositionY = (DisplayConstants.HEIGHT / 2) - (DisplayConstants.GOAL_HEIGHT / 2);
 
         return new Rectangle(DisplayConstants.FIELD_MARGIN - DisplayConstants.GOAL_WIDTH, (int) goalPositionY,
                 DisplayConstants.GOAL_WIDTH, (int) DisplayConstants.GOAL_HEIGHT);
     }
 
+    /**
+     * Method that calculates and returns the rectangle of the right goalm of the map.
```
Comment (bras1223) [inline Soccar [Client]/src/nl/soccar/util/MapUtilities.java]: - Method that calculates and returns the rectangle of the right goal**_m_** of the map.

File: Soccar [Client]/src/nl/soccar/util/MapUtilities.java [hunk 12]
```diff
@@ -9,13 +9,23 @@
...
+     * 
+     * @return The rectangle of the right goal of the map.
```
Comment (bras1223) [inline Soccar [Client]/src/nl/soccar/util/MapUtilities.java]: @return **Rectangle** The rectangle of the right goal of the map.

**Review by bras1223 — COMMENTED** [review]
"From" of "of"? Wie weet hier het antwoord op?
Verder kleine aanpassingen; Even bekijken of deze correct zijn.

**Review by bras1223 — APPROVED** [review]
Perfect!

**Pull request merged**